### ISSUE TITLE
Wrong redirect from SECURE connection to INSECURE

### ISSUE THREAD

**Issue by Nokim** [phpmyadmin/phpmyadmin#11412]
phpMyAdmin 4.4.13.1 (same on older versions)

When i try to open phpMyAdmin via HTTPS link it it opens but on login redirects me to HTTP. If i set $cfg['ForceSSL'] = true, i get infinite loop of redirects. Manual switch back to HTTPS (by opening right URL) works.
My setup:
1. Front-end gateway, handles SSL connection, redirect requests to working server. SSL end here. Used nginx.
2. Back-end server, process requests. phpMyAdmin here. Obviously no HTTPS here. Used nginx + php-fpm.
In such setup if phpMyAdmin didn't try to change protocol (say using relative links instead of absolute) all should be fine. Modern browsers support this https://en.wikipedia.org/wiki/HTTP_location

**Comment by madhuracj** [comment]
May be related to https://github.com/phpmyadmin/phpmyadmin/issues/11254?

**Comment by Nokim** [comment]
Yes, this case have exactly same origin and i understand why $cfg['ForceSSL'] can't work in my case (because of my front-end server, which is reverse proxy).

However this case can be easily solved by using relative URL for redirect instead of absolute which should be fine in case $cfg['ForceSSL'] = false. phpMyAdmin shouldn't change protocol if not instructed otherwise.

**Comment by Nokim** [comment]
Using $cfg['PmaAbsoluteUri'] = 'https://my.server/db/' seems to solve problem for me however i insist that phpMyAdmin shouldn't change protocol if not instructed otherwise. Updated HTTP 1.1 specifications (IETF RFC 7231) allow to use relative URL in redirects so it's easy.

**Comment by lem9** [comment]
In our doc we say "Users of port forwarding will need to set $cfg['PmaAbsoluteUri']" but I agree with you. I am starting a discussing on the developers list to suggest the new behavior and to replace the configuration directive.

**Comment by lem9** [comment]
On the list, I said:
````
This is related to [0]. According to [1] we should replace the
PmaAbsoluteUri directive with PmaUri, which would accept relative URLs.
Or maybe we don't need this directive anymore.

See mainly checkPmaAbsoluteUri().

It's a little late for 4.5.0 and even for 4.5.x, so I suggest we target
4.6 for this.
````
and @nijel replied:
````

> This is related to [0]. According to [1] we should replace the
> PmaAbsoluteUri directive with PmaUri, which would accept relative URLs.
> Or maybe we don't need this directive anymore.

If we decide to use relative locations, then there is no need for this
configuration directive ...

> See mainly checkPmaAbsoluteUri().

... and for piece of code
.
> It's a little late for 4.5.0 and even for 4.5.x, so I suggest we target
> 4.6 for this.

I think removing all this in 4.6 is good idea.
````

**Comment by nijel** [comment]
@ibennetch Any progress on this? While working on #11731 I noticed the code for PmaAbsoluteUri handling is one of candidates for removal :-).

**Comment by ibennetch** [comment]
I was working on it and got distracted with another project. I'd still like to work on it if it's not blocking your #11731 work.

**Comment by nijel** [comment]
It's not blocking (I can keep the code there as is), I was just curious about the state as I saw the code and remembered about discussion to remove it :-).

**Comment by ibennetch** [comment]
I'm going to remove my assignment for now; if someone else wishes to work on this it's fine. If not, I'll keep it in mind to revisit later.

**Comment by nijel** [comment]
I'll look into this. It will also fix https://github.com/phpmyadmin/docker/issues/12

**Comment by zaverichintan** [comment]
@ibennetch can I work on that issue, I am planning to apply for GSOC 2016.

**Comment by lem9** [comment]
@zaverichintan This issue is already assigned (see "assignee" above).

**Comment by kzalewski** [comment]
Confirmed fixed in 4.6-rc2.  Thank you.

**Comment by ibennetch** [comment]
Excellent! Thanks for the confirmation.
